src/ModelContextProtocol/McpEndpointExtensions.cs

```diff
@@ -165,4 +165,75 @@ public static Task NotifyProgressAsync(
+    /// <param name="endpoint">The endpoint issuing the notification.</param>
+    /// <param name="notification">The notification to send.</param>
+    /// <param name="cancellationToken">A token to cancel the operation.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="endpoint"/> is <see langword="null"/>.</exception>
+    /// <returns>A task representing the completion of the operation.</returns>
+    public static Task NotifyAsync(
+        this IMcpEndpoint endpoint,
+        JsonRpcNotification notification,
+        CancellationToken cancellationToken = default)
+    {
+        Throw.IfNull(endpoint);
+
+        return endpoint.SendMessageAsync(notification, cancellationToken);
+    }
+    
+    /// <summary>
+    /// Notifies the connected endpoint of an event.
+    /// </summary>
+    /// <param name="endpoint">The endpoint issuing the notification.</param>
+    /// <param name="method">The method to call.</param>
+    /// <param name="parameters">The parameters to send.</param>
+    /// <param name="cancellationToken">A token to cancel the operation.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="endpoint"/> is <see langword="null"/>.</exception>
+    /// <returns>A task representing the completion of the operation.</returns>
+    public static Task NotifyAsync(
+        this IMcpEndpoint endpoint,
+        string method,
+        JsonNode? parameters = null,
+        CancellationToken cancellationToken = default)
+    {
+        Throw.IfNull(endpoint);
+
+        return endpoint.NotifyAsync(new()
+        {
+            Method = method,
+            Params = parameters,
+        }, cancellationToken);
+    }
+
+    /// <summary>
+    /// Notifies the connected endpoint that a request has been cancelled.
+    /// </summary>
+    /// <param name="endpoint">The endpoint issuing the notification.</param>
+    /// <param name="requestId">The ID of the request to cancel.</param>
+    /// <param name="reason">An optional reason for the cancellation.</param>
+    /// <param name="cancellationToken">A token to cancel the operation.</param>
+    /// <returns>A task representing the completion of the operation.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="endpoint"/> is <see langword="null"/>.</exception>
+    public static Task NotifyCancelAsync(
+        this IMcpEndpoint endpoint,
+        RequestId requestId,
+        string? reason = null,
+        CancellationToken cancellationToken = default)
+    {
+        Throw.IfNull(endpoint);
+
+        return endpoint.SendMessageAsync(new JsonRpcNotification()
+        {
+            Method = NotificationMethods.CancelledNotification,
+            Params = JsonSerializer.SerializeToNode(new CancelledNotification
+            {
+                RequestId = requestId,
+                Reason = reason,
+            }, McpJsonUtilities.JsonContext.Default.CancelledNotification),
+        }, cancellationToken);
+    }
```

Comment:
I thought the intent of this PR was to automatically propagate cancellation to the server, but I no longer see the registration with the cancellation token to call SendMessageAsync.